aspnetcore/fundamentals/servers/kestrel.md

```diff
@@ -160,6 +155,34 @@ The Kestrel web server has constraint configuration options that are especially
 
 Set constraints on the <xref:Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerOptions.Limits> property of the <xref:Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerOptions> class. The `Limits` property holds an instance of the <xref:Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerLimits> class.
 
+### Keep-alive timeout
+
+<xref:Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerLimits.KeepAliveTimeout>
+
+Gets or sets the keep-alive timeout. Defaults to 2 minutes.
+
+::: moniker range=">= aspnetcore-2.2"
+
+[!code-csharp[](kestrel/samples/2.x/KestrelSample/Program.cs?name=snippet_Limits&highlight=15)]
+
+::: moniker-end
+
+::: moniker range="< aspnetcore-2.2"
+
+```csharp
+public static IWebHostBuilder CreateWebHostBuilder(string[] args) =>
+    WebHost.CreateDefaultBuilder(args)
+        .UseStartup<Startup>()
+        .UseKestrel(options =>
+        {
+            options.Limits.KeepAliveTimeout = TimeSpan.FromMinutes(2);
+        });
+```
+
+::: moniker-end
+
+For information on the ASP.NET Core Module `requestTimeout` when hosting in IIS, see <xref:host-and-deploy/aspnet-core-module#attributes-of-the-aspnetcore-element>.
```

Comment:
How is ANCM's requestTimeout at all related to Kestrel's KeepAlive timeout?

Reply:
I'm not allowed to ping u with questions on issues any longer (unless critical/follow-up). Some of these are *uneducated wild guesses* ... e.g., which IIS out-of-process ANCM limits are in effect for Kestrel ... so, we'll have to clean them up on PR review.

Reply:
Actually ... I retract that last remark. It's been a long day here. Some of this can only be considered in the morning after a restful night's sleep and a hot cup of ☕️.

Reply:
I don't quite follow the question. IIS out-of-proc has its own self-contained limits. Kestrel is always the server running behind out-of-proc, and has its own set of limits. The only special case I'm aware of when the two limits collide is with request body size limits.
https://github.com/aspnet/AspNetCore/blob/01a87989364dea1b18743a9fe3757eccc220b2ca/src/Servers/IIS/IISIntegration/src/IISMiddleware.cs#L115-L120
IIS out-of-proc disables Kestrel's request body size limits, they're redundant.

Reply:
> I don't quite follow the question. 

Withdrawn remark/question. It's late here ... I'm a bit brain-fried this evening.

> only special case I'm aware of when the two limits collide is with request body size limits

Cool. Let me get a night's :zzz: 🛌, and I'll see if I can get this right in the docs in the morning.